blogp3.md

```diff
@@ -0,0 +1,434 @@
+
+```
+create-react-app client
+cd client
+```
+
+You should now have two folders: server and client. 
+
+Now you can run these commands to install your dependencies and start running your app.
+
+```
+npm install
+npm start
+```
+
+Navigate to `http://localhost:3000` to see if your app is up and running. It should look something like this:
+
+[![startingreactpage](https://github.com/natalieh235/songrecproject/raw/blog/images/startingreactpage.png)](https://github.com/natalieh235/songrecproject/blob/blog/images/startingreactpage.png)
+
+#### Step 2- Create the Homepage
+
+So this React app will involve some very simple routing- basically this means that different urls will render different pages. For example, going to `localhost:3000/home` will take you to a homepage, while going to `localhost:3000/song` will perhaps show a list of songs. We're going to focus on creating a basic homepage, which will look something like this:
+
+[![reacthomepage](https://github.com/natalieh235/songrecproject/raw/blog/images/reacthomepage.png)](https://github.com/natalieh235/songrecproject/blob/blog/images/reacthomepage.png)
+
+Get started by making a new file called `Home.js`. I stuck this inside a folder `components`, since we're going to end up making a lot more files later on.
+
+Add in this code:
+
+```
+import React from 'react';
+
+function Home() {
+  return (
+    <div className="home-page">
+        <h1>Upload your face. We pick the songs.</h1>
+        <a href="https://localhost:8888">
+          <button className="green-btn">Login with Spotify</button>
+        </a>   
+    </div>
+  );
+}
+
+export default Home;
+```
+
+This is a simple functional component, and simply contains a header and button. Notice that the button is redirecting us to our OAuth server page.
+
+Return to your React app page and check to make sure the button takes you to the OAuth page, but don't log in quite yet.
+
+#### Step 2- Creating the Redirect Page
+
+After the user logs in, we need to redirect them to a different page in our React app. To do this, I'm creating a new file called `UploadFace.js`. Note: this name ends up being kind of misleading, since this ends being the page where all the uploading and song-finding happens, but just bear with me.
+
+Copy paste this code into UploadFace:
+
+This creates a class component with state variables `token` and `userInfo`. Once the component mounts, we're using the lifecycle method `componentDidMount()` to immediately call the `getHashParams` function. The `getHashParams` function is actually used in the Spotify Oauth example code to parse the query string and extract the token. We're stealing this function to get the access token in our client-side.
+
+Once we get the token, we set our state variable `token` and make a API call using the token to get the user's display name and id. To test that this actually works, let's render a "Logged in as {display_name}" statement. This will change later.
```

Comment:
```suggestion
Once we get the token, we set our state variable `token` and make an API call using the token to get the user's display name and id. To test that this works, let's render a "Logged in as {display_name}" statement. This will change later.
```